modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -34,25 +35,98 @@ PolygonMeshGeneratorBase::generate()
+    const std::vector<Real> ring_inner_boundary_layer_fractions,
+    const std::vector<unsigned int> ring_inner_boundary_layer_intervals,
+    const std::vector<Real> ring_inner_boundary_layer_biases,
+    const std::vector<Real> ring_outer_boundary_layer_fractions,
+    const std::vector<unsigned int> ring_outer_boundary_layer_intervals,
+    const std::vector<Real> ring_outer_boundary_layer_biases,
+    std::vector<Real> ducts_center_dist,
+    const std::vector<unsigned int> ducts_layers,
+    const std::vector<Real> duct_radial_biases,
+    const std::vector<Real> duct_inner_boundary_layer_fractions,
+    const std::vector<unsigned int> duct_inner_boundary_layer_intervals,
+    const std::vector<Real> duct_inner_boundary_layer_biases,
+    const std::vector<Real> duct_outer_boundary_layer_fractions,
+    const std::vector<unsigned int> duct_outer_boundary_layer_intervals,
+    const std::vector<Real> duct_outer_boundary_layer_biases,
+    bool has_rings,
+    bool has_ducts,
+    const Real pitch,
+    const unsigned int num_sectors_per_side,
+    const unsigned int background_intervals,
+    const Real background_radial_bias,
+    const Real background_inner_boundary_layer_width,
+    const unsigned int background_inner_boundary_layer_intervals,
+    const Real background_inner_boundary_layer_bias,
+    const Real background_outer_boundary_layer_width,
+    const unsigned int background_outer_boundary_layer_intervals,
+    const Real background_outer_boundary_layer_bias,
+    dof_id_type & node_id_background_meta,
+    const unsigned int side_number,
+    const unsigned int side_index,
+    const std::vector<Real> azimuthal_tangent,
+    const subdomain_id_type block_id_shift,
+    const bool quad_center_elements,
+    const boundary_id_type boundary_id_shift)
 {
   auto mesh = buildReplicatedMesh(2);
 
+  // Calculate biasing terms
+  // background region needs to be split into three parts
+  const auto main_background_bias_terms =
+      biasTermsCalculator(background_radial_bias, background_intervals);
+  const auto inner_background_bias_terms = biasTermsCalculator(
+      background_inner_boundary_layer_bias, background_inner_boundary_layer_intervals);
+  const auto outer_background_bias_terms = biasTermsCalculator(
+      background_outer_boundary_layer_bias, background_outer_boundary_layer_intervals);
+  auto rings_bias_terms = biasTermsCalculator(ring_radial_biases,
+                                              rings,
+                                              ring_inner_boundary_layer_fractions,
+                                              ring_inner_boundary_layer_intervals,
+                                              ring_inner_boundary_layer_biases,
+                                              ring_outer_boundary_layer_fractions,
+                                              ring_outer_boundary_layer_intervals,
+                                              ring_outer_boundary_layer_biases);
```

Comment:
this makes me think the struct could even hold 6 members? Not sure it's true though, background has less params?